File: hw/chicaiza/u3/HW22Strategy/sortapp/src/main/java/ec/edu/espe/sort/model/InsertionSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.sort.model;

/**
 *
 * @author Chicaiza Michael
 */
public class InsertionSort implements SortingStrategy{
    public int[] sort(int data[]){
        int[] result = data.clone();

        for (int i = 1; i < result.length; i++) {
            int key = result[i];
            int j = i - 1;

            while (j >= 0 && result[j] > key) {
                result[j + 1] = result[j];
                j--;
            }
            result[j + 1] = key;
            }

        System.out.println("Using InsertionSort");
        return result;
        
    }
    
}

File: hw/fuentes/U3/HW22STRATEGY/ExamU3Team3/SortAppPY/ec/edu/espe/clientApp/model/insertion_sort.py
from model.sorting_strategy import SortingStrategy

class InsertionSort(SortingStrategy):
    def sort(self, data: list[int]) -> list[int]:
        result = data.copy()
        for i in range(1, len(result)):
            key = result[i]
            j = i - 1
            while j >= 0 and result[j] > key:
                result[j + 1] = result[j]
                j -= 1
            result[j + 1] = key
        print("Using InsertionSort")
        return result
